spacelift/data_worker_pools.go

```diff
@@ -0,0 +1,94 @@
+package spacelift
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/structs"
+)
+
+func dataWorkerPools() *schema.Resource {
+	return &schema.Resource{
+		Read: dataWorkerPoolsRead,
+		Schema: map[string]*schema.Schema{
+			"worker_pools": &schema.Schema{
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"worker_pool_id": {
+							Type:        schema.TypeString,
+							Description: "ID of the worker pool",
+							Required:    true,
+						},
+						"config": {
+							Type:        schema.TypeString,
+							Description: "credentials necessary to connect WorkerPool's workers to the control plane",
+							Sensitive:   true,
+							Required:    true,
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Description: "name of the worker pool",
+							Required:    true,
+						},
+						"description": {
+							Type:        schema.TypeString,
+							Description: "description of the worker pool",
+							Required:    true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataWorkerPoolsRead(d *schema.ResourceData, meta interface{}) error {
+	var query struct {
+		WorkerPools []*structs.WorkerPool `graphql:"workerPools()"`
+	}
+	variables := map[string]interface{}{}
+
+	if err := meta.(*internal.Client).Query(&query, variables); err != nil {
+		return errors.Wrap(err, "could not query for worker pools")
+	}
+
+	workerPools := query.WorkerPools
+	if workerPools == nil {
```

Comment:
Quite sure this check is unnecessary. I suppose it's ok if the list is empty?

Reply:
I would say that returning early would be better way to handle the scenario when there are no worker pools?

Reply:
Returning early should be OK, but returning an error isn't - having no worker pools is a perfectly valid (though weird) scenario, so no error necessary here.